pkgs/applications/blockchains/nimbus/default.nix

```diff
@@ -0,0 +1,81 @@
+{
+  stdenv, fetchgit, lib, which, writeScriptBin,
+  # Options: nimbus_light_client, nimbus_validator_client, nimbus_signing_node
+  makeTargets ? [ "nimbus_beacon_node" ],
+  # WARNING: CPU optmizations that make binary not portable.
+  nativeBuild ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nimbus";
+  version = "22.10.0";
+  commit = "2d6911d5";
+  name = "${pname}-${version}-${commit}";
+
+  src = fetchgit {
+    url = "https://github.com/status-im/nimbus-eth2.git";
+    rev = "v${version}";
+    sha256 = "sha256-+rkBNop1HDhfdncmlP9UgA6OC4t6SgDhfbb3wZ1INnY=";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = let
+    # Fix for Nim compiler calling 'git rev-parse' and 'lsb_release'.
+    fakeGit = writeScriptBin "git" "echo $commit";
+    fakeLsbRelease = writeScriptBin "lsb_release" "echo nix";
+  in [ fakeGit fakeLsbRelease which ];
+
+  enableParallelBuilding = true;
+
+  NIMFLAGS = lib.optionalString (!nativeBuild) " -d:disableMarchNative";
+
+  makeFlags = makeTargets;
+
+  preBuildPhases = [ "buildCompiler" ];
+
+  # Generate vendor/.nimble contents with correct paths.
+  configurePhase = ''
+    export EXCLUDED_NIM_PACKAGES=""
+    export NIMBLE_LINK_SCRIPT=$PWD/vendor/nimbus-build-system/scripts/create_nimble_link.sh
+    export NIMBLE_DIR=$PWD/vendor/.nimble
+    export PWD_CMD=$(which pwd)
+    patchShebangs scripts > /dev/null
+    patchShebangs $PWD/vendor/nimbus-build-system/scripts > /dev/null
+    for dep_dir in $(find vendor -type d -maxdepth 1); do
+        pushd "$dep_dir" >/dev/null
+        $NIMBLE_LINK_SCRIPT "$dep_dir"
+        popd >/dev/null
+    done
+  '';
+
+  # Nimbus uses it's own specific Nim version bound as a Git submodule.
+  buildCompiler = ''
+    # Necessary for nim cache creation
+    export HOME=$PWD
+    make -j$NIX_BUILD_CORES build-nim
+    export PATH="$PWD/vendor/nimbus-build-system/vendor/Nim/bin:$PATH"
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    rm build/generate_makefile
+    cp build/* $out/bin
+  '';
+
+  meta = with lib; {
+    homepage = "https://nimbus.guide/";
+    downloadPage = "https://github.com/status-im/nimbus-eth2/releases";
+    changelog = "https://github.com/status-im/nimbus-eth2/blob/stable/CHANGELOG.md";
+    description = "Nimbus is a lightweight client for the Ethereum consensus layer";
+    longDescription = ''
+      Nimbus is an extremely efficient consensus layer client implementation.
+      While it's optimised for embedded systems and resource-restricted devices --
+      including Raspberry Pis, its low resource usage also makes it an excellent choice
+      for any server or desktop (where it simply takes up fewer resources).
+    '';
+    branch = "stable";
+    license = with licenses; [ asl20 mit ];
+    maintainers = with maintainers; [ jakubgs ];
+    platforms = with platforms; x86_64 ++ arm ++ aarch64;
```

Comment:
Is there a reason to limit this to only these three architectures?

Reply:
Yes. I discussed this with the team and those are the only ones we test for and build in CI. So we have no guarantee it would work in a stable manner on any others. And considering this software can lose you money if it's not stable it's kinda important to indicate which architectures we can guarantee to work well.

Reply:
Also for that reason these are the only binaries we include in releases:
https://github.com/status-im/nimbus-eth2/releases/tag/v22.10.1

Reply:
Then this is not quite right. platforms.arm is an alias to the following ``[ "armv7a-darwin" "armv5tel-linux" "armv6l-linux" "armv7a-linux" "armv7l-linux" "armv6l-netbsd" "armv7a-netbsd" "armv7l-netbsd" "arm-none" "armv6l-none" ]`` and platforms.x85_64 is one to ``[ "x86_64-cygwin" "x86_64-darwin" "x86_64-freebsd13" "x86_64-genode" "x86_64-solaris" "x86_64-linux" "x86_64-netbsd" "x86_64-none" "x86_64-openbsd" "x86_64-redox" "x86_64-windows" ]``.

but in nixpkgs we do not generally limit packages to the exact platforms anyway because mostly only linux and darwin matter. I would suggest to set this to platforms.linux.

Reply:
But we don't run just on Linux, we run on Windows and Darwin(macOS). And we test only specific architectures for those platforms, and I don't think we can confidently say that this software works well on other platforms we do not test in CI, and for that reason we should inform people which platforms are stable.

Considering this software can lose you money, I'd say this is kinda important.

Reply:
I can try listing specific platform and architecture combinations instead.